dask_geopandas/geohash.py

```diff
@@ -0,0 +1,221 @@
+import numpy as np
+import pandas as pd
+from .utils import _calculate_mid_points
+
+
+# Implementation based on deprecated https://pypi.org/project/neathgeohash/#description
+
+
+def _geohash(gdf, precision):
+    """
+    Calculate geohash based on the middle points of the geometry bounds
+    for a given precision
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+    precision : int
+        precision of the Geohash
+
+    Returns
+    ---------
+    type : pandas.Series
+        Series containing geohash
+    """
+
+    # Calculate bounds
+    bounds = gdf.bounds.to_numpy()
+    # Calculate mid points based on bounds
+    x_mids, y_mids = _calculate_mid_points(bounds)
+    # Return coordinates
+    coords = np.array([y_mids, x_mids]).T
+    # Encode coords with Geohash
+    geohash = encode_geohash(coords, precision)
+
+    return pd.Series(geohash, index=gdf.index, name="geohash")
+
+
+def encode_geohash(coords, precision):
+    """
+    Calculate geohash based on coordinates for a
+    given precision
+
+    Parameters
+    ----------
+    coords : array_like
+    precision : int
+        precision of the Geohash
+
+    Returns
+    ---------
+    geohash: array containing geohashes for each coordinate
+    """
+
+    quantized_coords = _encode_quantize_points(coords)
+    g_uint64 = _encode_into_uint64(quantized_coords)
+    gs_uint8_mat = _encode_base32(g_uint64)
+    geohash = _encode_unicode(gs_uint8_mat, precision)
+
+    return geohash
+
+
+def _encode_quantize_points(coords):
+    """
+    Quantize coordinates by mapping
+    unit intervals [0, 1] & multiplying by 2^32.
+
+    Parameters
+    ----------
+    coords : array_like
+        coordinate pairs
+
+    Returns
+    ---------
+    quantized_coords : array_like
```

Comment:
```suggestion
    array_like of shape (n, 2)
```

I don't think you need to name the output values.